selftests/unit/test_utils_asset.py

```diff
@@ -49,9 +53,45 @@ def testFetch_location(self):
             content = f.read()
         self.assertEqual(content, expected_content)
 
+    def testFecth_expire(self):
+        foo_tarball = asset.Asset(self.assetname,
+                                  asset_hash=self.assethash,
+                                  algorithm='sha1',
+                                  locations=[self.url],
+                                  cache_dirs=[self.cache_dir],
+                                  expire=None).fetch()
+        ctime = os.lstat(foo_tarball)[stat.ST_CTIME]
+        time.sleep(1)
+        asset.Asset(self.assetname,
+                    asset_hash=self.assethash,
+                    algorithm='sha1',
+                    locations=[self.url],
+                    cache_dirs=[self.cache_dir],
+                    expire=0).fetch()
+        new_ctime = os.lstat(foo_tarball)[stat.ST_CTIME]
+        self.assertNotEqual(ctime, new_ctime)
+
+    def testFecth_not_expire(self):
+        foo_tarball = asset.Asset(self.assetname,
+                                  asset_hash=self.assethash,
+                                  algorithm='sha1',
+                                  locations=[self.url],
+                                  cache_dirs=[self.cache_dir],
+                                  expire=None).fetch()
+        ctime = os.lstat(foo_tarball)[stat.ST_CTIME]
+        asset.Asset(self.assetname,
+                    asset_hash=self.assethash,
+                    algorithm='sha1',
+                    locations=[self.url],
+                    cache_dirs=[self.cache_dir],
+                    expire=-1).fetch()
+        new_ctime = os.lstat(foo_tarball)[stat.ST_CTIME]
+        self.assertEqual(ctime, new_ctime)
```

Comment:
well unless the second turns over this will be equal even when the file is re-created. So you could either use `time.sleep`s to make sure new second started, but you can also use `flexmock` to mock the returns of `time.time()` and use `st_ino` to see whether it changed (which it should if the file is re-created)